modules/ossm-upgrade-considerations.adoc

```diff
@@ -0,0 +1,28 @@
+// Module included in the following assemblies:
+// * service_mesh/v2x/upgrading-ossm.adoc
+
+[id="ossm-upgrade-considerations_{context}"]
+= Upgrade considerations
+
+The `maistra.io/` label or annotation should not be used on a user-created custom resource, because it indicates that the resource was generated by and should be managed by the {ProductName} Operator.
+
+[WARNING]
+====
+During the upgrade, the Operator will make changes, including deleting or replacing files, to resources that include the following labels or annotations that indicate that the resource is managed by the Operator.
+====
+
+Before upgrading you should check for user-created custom resources that include the following labels or annotations :
+
+* `maistra.io/`  AND the `app.kubernetes.io/managed-by` label set to `maistra-istio-operator` ({ProductName})
+
+* `kiali.io/` (Kiali)
+
+* `jaegertracing.io/` ({JaegerName})
+
+* `logging.openshift.io/` (Red Hat Elasticsearch)
+
+Before upgrading, you should check your user-created custom resources for labels or annotations that indicate they are Operator managed.  Remove the label or annotation from custom resources that you do not want to be managed by the Operator.
+
+When upgrading to version 2.0, the Operator will only delete resources with these labels in the same namespace as the SMCP.
+
+When upgrading to version 2.1, the Operator will delete resources with these labels in member namespaces.
```

Comment:
The operator will delete these resources in all namespaces, not just member namespaces.

Reply:
Also, the operator deletes these resources not only when you upgrade the operator, but any time you update the SMCP object. It does this because the labels indicate that the resource is managed by the operator, but the resource shuldn't exist according to the current SMCP.spec. Basically, any time you change the SMCP.spec, the operator generates a bunch of resources, applies them to the cluster, and then removes any resources that weren't generated by the operator for this SMCP.spec update.

Reply:
Since this topic applies specifically to upgrade, I've captured this comment in OSSMDOC-489.